prometheus/src/main/scala/filodb/prometheus/parse/Parser.scala

```diff
@@ -15,8 +19,52 @@ trait BaseParser extends Expressions with JavaTokenParsers with RegexParsers wit
     "[a-zA-Z_:][a-zA-Z0-9_:\\-\\.]*".r ^^ { str => Identifier(str) }
   }
 
-  protected lazy val labelValueIdentifier: PackratParser[Identifier] =
-    "([\"'])(?:\\\\\\1|.)*?\\1".r ^^ { str =>  Identifier(str.substring(1, str.size-1)) } //remove quotes
+  protected lazy val labelValueIdentifier: PackratParser[Identifier] = {
+    // Parse a quoted identifier, supporting escapes, with quotes removed.
+    new PackratParser[Identifier]() {
+      def apply(in: Input): ParseResult[Identifier] = {
+        val source = in.source
+        var offset = in.offset
+
+        (whiteSpace findPrefixMatchOf (source.subSequence(offset, source.length))) match {
+          case Some(matched) => offset += matched.end
+          case None =>
+        }
+
+        val quote = source.charAt(offset); offset += 1
+
+        if (quote != '\'' && quote != '"') {
+          return Failure("quote character expected", in)
+        }
+
+        val bob = new StringBuilder()
+
+        while (offset < source.length) {
+          var c = source.charAt(offset); offset += 1
+
+          if (c == quote) {
+            return Success(Identifier(bob.toString()), in.drop(offset - in.offset))
```

Comment:
How is a string like `"dfsd"df` treated ?

Reply:
It should stop the identifier parsing at the quote, and the trailing df is handled by the next parsing step. It will likely fail.